warp10/src/main/java/io/warp10/script/processing/Pencode.java

```diff
@@ -39,36 +34,47 @@
 
 import com.sun.imageio.plugins.png.PNGMetadata;
 
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+import io.warp10.script.functions.TYPEOF;
+import processing.core.PImage;
+
 /**
  * Encode the PGraphics on the stack into a base64 string suitable for a data URL.
  */
 public class Pencode extends NamedWarpScriptFunction implements WarpScriptStackFunction {
-  
+
   public Pencode(String name) {
     super(name);
   }
-  
+
   @Override
   public Object apply(WarpScriptStack stack) throws WarpScriptException {
     Object top = stack.pop();
 
     Map<Object,Object> chunks = null;
-    
+
     if (top instanceof Map) {
       chunks = (Map<Object,Object>) top;
       top = stack.pop();
     }
-    
-    if (!(top instanceof processing.core.PGraphics)) {
-      throw new WarpScriptException(getName() + " operates on a PGraphics instance.");
+
+    if (!(top instanceof processing.core.PImage)) {
```

Comment:
You could use `PImage` instead of `processing.core.PImage`.
Same with `PGraphics` and `processing.core.PGraphics` x3.